src/System.ServiceModel.Primitives/tests/ServiceModel/ChannelFactoryTest.cs

```diff
@@ -231,4 +231,41 @@ public static void ChannelFactory_Async_Open_Close()
             }
         }
     }
+
+    [Theory]
+    [InlineData(true)]
+    [InlineData(false)]
+    public static void ChannelFactory_AllowCookies(bool allowCookies)
+    {
+        ChannelFactory<IWcfService> factory = null;
+
+        try
+        {
+            factory = new ChannelFactory<IWcfService>(
+                     new BasicHttpBinding()
+                     {
+                         AllowCookies = allowCookies
+                     },
+                     new EndpointAddress(FakeAddress.HttpAddress));
+
+            IWcfService serviceProxy = factory.CreateChannel();
+
+            IHttpCookieContainerManager cookieManager = ((IChannel)serviceProxy).GetProperty<IHttpCookieContainerManager>();
+            Assert.True(allowCookies == (cookieManager != null),
+                $"AllowCookies was {allowCookies}, 'cookieManager != null' was expected to be {allowCookies}, but it was {cookieManager != null}.");
+
+            if (allowCookies)
+            {
+                Assert.True(allowCookies == (cookieManager.CookieContainer != null),
+                    $"AllowCookies was {allowCookies}, 'cookieManager.CookieContainer != null' was expected to be {allowCookies}, but it was {cookieManager != null}.");
```

Comment:
Same comment as before - would like to see `"AllowCookies was '{allowCookies}', 'cookieManager.CookieContainer != null' was expected to be '{allowCookies}' ..."